lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js

```diff
@@ -0,0 +1,487 @@
+* var ISAVE = new Int32Array( [ 3 ] );
+*
+* isaveIsThree( 4, X, 1, 0, V, 1, 0, EST, 0, KASE, 0, ISGN, 1, 0, ISAVE, 0 );
+*
+* // EST => <Float64Array>[ 9.0 ]
+* // X => <Float64Array>[ -1.0, 1.0, -1.0, -1.0 ]
+* // ISGN => <Int32Array>[ -1, 1, -1, -1 ]
+* // KASE => <Int32Array>[ 2 ]
+* // ISAVE => <Int32Array>[ 4 ]
+* // V => <Float64Array>[ -2.0, 0.0, 3.0, -4.0 ]
+*/
+function isaveIsThree( N, X, strideX, offsetX, V, strideV, offsetV, EST, offsetEST, KASE, offsetKASE, ISGN, strideISGN, offsetISGN, ISAVE, offsetISAVE ) {
+	var estold;
+	var ix1;
+	var is1;
+	var ix;
+	var is;
+	var xs;
+	var i;
+	var j;
+
+	dcopy( N, X, strideX, offsetX, V, strideV, offsetV );
+	estold = EST[ offsetEST ];
+	EST[ offsetEST ] = dasum( N, V, strideV, offsetV );
+
+	ix = offsetX;
+	is = offsetISGN;
+	for ( i = 0; i < N; i++ ) {
+		if ( X[ ix ] >= 0.0 ) {
+			xs = 1.0;
+		} else {
+			xs = -1.0;
+		}
+
+		if ( nint( xs ) !== ISGN[ is ] ) {
+			if ( EST[ offsetEST ] <= estold ) {
+				finalStep( N, X, strideX, offsetX, KASE, offsetKASE, ISAVE, offsetISAVE );
+				return;
+			}
+
+			ix1 = offsetX;
+			is1 = offsetISGN;
+			for ( j = 0; j < N; j++ ) {
+				if ( X[ ix1 ] >= 0.0 ) {
+					X[ ix1 ] = 1.0;
+				} else {
+					X[ ix1 ] = -1.0;
+				}
+				ISGN[ is1 ] = nint( X[ ix1 ] );
+
+				ix1 += strideX;
+				is1 += strideISGN;
+			}
+			KASE[ offsetKASE ] = 2;
+			ISAVE[ offsetISAVE ] = 4;
+			return;
+		}
+
+		ix += strideX;
+		is += strideISGN;
+	}
+
+	finalStep( N, X, strideX, offsetX, KASE, offsetKASE, ISAVE, offsetISAVE );
+}
+
+/**
+* Attempts to refine the estimate by cycling through multiple basis vectors.
+*
+* @private
+* @param {PositiveInteger} N - number of rows/columns in `A`
+* @param {Float64Array} X - input/output vector having `N` indexed elements, contains the current or next matrix-vector product
+* @param {integer} strideX - stride length for `X`
+* @param {NonNegativeInteger} offsetX - starting index for `X`
+* @param {Int32Array} ISAVE - integer array having 3 indexed elements, used internally to maintain state across multiple calls
+* @param {integer} strideISAVE - stride length for `ISAVE`
+* @param {NonNegativeInteger} offsetISAVE - starting index for `ISAVE`
+* @param {Int32Array} KASE - single-element array that controls the reverse communication.
+* @param {NonNegativeInteger} offsetKASE - starting index for `KASE`
+* @returns {void}
+*
+* @example
+* var X = new Float64Array( [ 2.0, 0.5, -3.0, 4.0 ] );
+* var ISAVE = new Int32Array( [ 2, 1, 3 ] );
+* var KASE = new Int32Array( [ -1 ] );
+*
+* isaveIsFour( 4, X, 1, 0, ISAVE, 1, 0, KASE, 0 );
+*
+* // X => <Float64Array>[ 0.0, 0.0, 0.0, 1.0 ]
+* // ISAVE => <Int32Array>[ 3, 3, 4 ]
+* // KASE => <Int32Array>[ 1 ]
+*/
+function isaveIsFour( N, X, strideX, offsetX, ISAVE, strideISAVE, offsetISAVE, KASE, offsetKASE ) {
+	var jlast;
+	var ix;
+	var i;
+
+	jlast = ISAVE[ offsetISAVE + strideISAVE ];
+	ISAVE[ offsetISAVE + strideISAVE ] = idamax( N, X, strideX, offsetX );
+
```

Comment:
I suggest making your expressions a bit tighter using temporary variables here for indices.

```javascript
var ii1 = offsetISAVE + strideISAVE;
var ii2 = ii1 + strideISAVE;

// ...

ISAVE[ ii1 ] && ISAVE[ ii2 ]
```